statistician.py

```diff
@@ -1,209 +1,114 @@
-        """
-        Private helper method used to update stats related with stats about methods used in game
-        :param bot1_method: String with method used by BOT1
-        :param bot2_method: String with method used by BOT2
-        """
-        bot1 = self.__bank[self.__bot1_name]
-        bot2 = self.__bank[self.__bot2_name]
-        bot1.methods_used[bot1_method] += 1
-        bot2.methods_used[bot2_method] += 1
-
-        for method in rules.methodsAsStrings:
-            self.__bank[GENERAL].methods_used[method] = bot1.methods_used[method] + bot2.methods_used[method]
-
-        self.__bank[GENERAL].methods_used_per_round.append(copy(self.__bank[GENERAL].methods_used))
-        bot1.methods_used_per_round.append(copy(bot1.methods_used))
-        bot2.methods_used_per_round.append(copy(bot2.methods_used))
-
-    def updateStats(self, bot1_method, bot2_method, winner, advantage):
-        """
-        Public method used to update all statistics gathered in game
-        :param bot1_method: String with method used by BOT1
-        :param bot2_method: String with method used by BOT2
-        :param winner: RoundWinner from enumeration package with information about round winner
-        :param advantage:
-        """
-        self.__round += 1
-        self.__updateRoundWonBy(winner)
-        self.__updateAdvantageAfterRounds(advantage)
-        self.__updateMethodsUsed(bot1_method, bot2_method)
-        self.__updateWonLostByMethod(bot1_method, bot2_method, winner)
-        self.__updatePointsEarnedWithMethod(bot1_method, bot2_method, winner)
-
-    def dumpStatistics(self):
-        """
-        Public method used to trigger saving all gathered data to files
-        """
-        path = f"./history/games/{self.__game_id}/stats"
-        Path(path).mkdir(parents=True, exist_ok=True)
-        bot1 = self.__bank[self.__bot1_name]
-        bot2 = self.__bank[self.__bot2_name]
-        general = self.__bank[GENERAL]
-
-        for method in rules.methodsAsStrings:
-            general.won_with_methods[method] = bot1.won_with_methods[method] + bot2.won_with_methods[method]
-            general.lost_with_methods[method] = bot1.lost_with_methods[method] + bot2.lost_with_methods[method]
-            general.points_earned_with_method[method] = bot1.points_earned_with_method[method] + \
-                                                        bot2.points_earned_with_method[method]
-
-        for name, data in self.__stats.items():
-            with open(f'{path}/{name}.json', 'w') as file:
-                file.writelines(json.dumps(data, sort_keys=True, indent=4))
+        with open('./history/leaderboard.html', 'w') as file:
+            file.writelines(leaderboard_html)
+
+    @staticmethod
+    def generateHTMLs():
+        cmd = f"jupyter nbconvert --execute --to html --no-input " \
+              f"--output-dir ./history/games/{db().getLastGameID} --output report.html " \
+              f"./history/insight.ipynb"
```

Comment:
Just use """ (triple quotation marks) to make multi line string.

Reply:
Nope. This is cmd generation. With triple quotion marks additional new line marks (\n) will be added to command. This will break process.